core/src/integration/java/com/radixdlt/integration/targeted/consensus/DelayAfterFormingTcAndFormAQcTest.java

```diff
@@ -62,104 +62,94 @@
  * permissions under this License.
  */
 
-package com.radixdlt.consensus.bft;
-
-import com.radixdlt.consensus.QuorumCertificate;
-import com.radixdlt.consensus.TimeoutCertificate;
-import java.util.Objects;
-
-/** The result of a round/round voting (either QC or TC). */
-public sealed interface RoundVotingResult {
-
-  static FormedQC qc(QuorumCertificate qc) {
-    return new FormedQC(qc);
-  }
-
-  static FormedTC tc(TimeoutCertificate tc) {
-    return new FormedTC(tc);
+package com.radixdlt.integration.targeted.consensus;
+
+import static com.radixdlt.harness.predicates.NodesPredicate.anyAtExactlyStateVersion;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.consensus.EpochNodeWeightMapping;
+import com.radixdlt.consensus.Proposal;
+import com.radixdlt.consensus.bft.Round;
+import com.radixdlt.environment.deterministic.network.MessageMutator;
+import com.radixdlt.environment.deterministic.network.MessageSelector;
+import com.radixdlt.harness.deterministic.DeterministicTest;
+import com.radixdlt.harness.deterministic.PhysicalNodeConfig;
+import com.radixdlt.modules.FunctionalRadixNodeModule;
+import com.radixdlt.modules.StateComputerConfig;
+import com.radixdlt.monitoring.Metrics;
+import org.junit.Test;
+
+// spotless:off
+/**
+ * Test a scenario where a timeout cert is formed first, but its
+ * processing is delayed, then a QC is formed
+ * which results in a QC output (RoundQuorumReached event with a QC).
```

Comment:
Nice big comment! I think it's missing a couple of key part to be complete though:
* We want to form a QC where we're able to, to try to ensure we can still form a 3-chain.
* This means that a single byzantine actor in the validator set can't force the formation of a TC instead of a QC, by sending out a timeout vote on some random vertex.